main.cpp

```diff
@@ -444,6 +457,16 @@ std::basic_string<T> uppercase(const std::basic_string<T>& s)
 
 clipp::formatting_ostream<std::ostream> fos(std::cout);
 
+static void sleep_ms(int ms) {
+#ifdef __unix__
```

Comment:
This define doesn't work on MacOSX.
https://stackoverflow.com/a/16107549 suggests that it ought to be:
```
#if defined(__unix__) || defined(__unix) || \
        (defined(__APPLE__) && defined(__MACH__))
```
but I guess it probably makes more sense to stay consistent with lines 2019 and 2020 and use:
```
#if defined(__linux__) || defined(__APPLE__)
```
?

P.S. Just out of curiosity I checked the `pico-sdk` repo, and that uses an inconsistent mixture of methods:
```
$ git grep "linux\|unix\|APPLE"
src/host/hardware_timer/timer.c:#if defined(__unix__) || defined(__APPLE__)
src/host/hardware_timer/timer.c:#if defined(__unix__) || defined(__APPLE__)
src/host/hardware_timer/timer.c:#if defined(__unix__) || defined(__APPLE__)
src/host/hardware_timer/timer.c:#if defined(__unix__)
src/host/hardware_uart/uart.c:#if defined(__unix) || defined(__APPLE__)
src/host/pico_platform/include/pico/platform.h:#ifdef __unix__
```

Reply:
the `timer.c` ones are correct; the `platform.h` one seems fine, because it seems to be include anyway on apple (or maybe by Clang)

Reply:
i fixed the macos build